livestream-compose-sample/src/main/java/io/getstream/livestream/compose/LiveStreamScreen.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.compose.handlers.SystemBackPressedHandler
+import io.getstream.chat.android.compose.state.messages.Thread
+import io.getstream.chat.android.compose.ui.messages.composer.MessageComposer
+import io.getstream.chat.android.compose.ui.messages.composer.components.MessageInput
+import io.getstream.chat.android.compose.ui.messages.header.MessageListHeader
+import io.getstream.chat.android.compose.ui.messages.list.MessageList
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.messages.MessageComposerViewModel
+import io.getstream.chat.android.compose.viewmodel.messages.MessageListViewModel
+
+
+@Composable
+fun LiveStreamScreen(
+    composerViewModel: MessageComposerViewModel,
+    listViewModel: MessageListViewModel,
+    onBackPressed: () -> Unit = {}
+) {
+    val currentState = listViewModel.currentMessagesState
+    val messageMode = listViewModel.messageMode
+    val isNetworkAvailable by listViewModel.isOnline.collectAsState()
+    val user by listViewModel.user.collectAsState()
+    LaunchedEffect(Unit) {
+        listViewModel.start()
+    }
+
+    val backAction = {
+        val isInThread = listViewModel.isInThread
+        val isShowingOverlay = listViewModel.isShowingOverlay
+
+        when {
+            isShowingOverlay -> listViewModel.selectMessage(null)
+            isInThread -> {
+                listViewModel.leaveThread()
+                composerViewModel.leaveThread()
+            }
+            else -> onBackPressed()
+        }
+    }
+
+    SystemBackPressedHandler(isEnabled = true, onBackPressed = backAction)
+
+    ChatTheme(isInDarkMode = true) {
+        Box(modifier = Modifier.fillMaxSize()) {
+            Scaffold(
+                modifier = Modifier.fillMaxSize(),
+                topBar = {
+                    MessageListHeader(
+                        modifier = Modifier
+                            .height(56.dp),
+                        channel = listViewModel.channel,
+                        currentUser = user,
+                        isNetworkAvailable = isNetworkAvailable,
+                        messageMode = messageMode,
+                        onBackPressed = backAction,
+                        onHeaderActionClick = {}
+                    )
+                },
+                bottomBar = {
+                    MyCustomComposer(composerViewModel)
+                }
+            ) {
+                MessageList(
+                    modifier = Modifier
+                        .padding(it)
+                        .fillMaxSize()
+                        .background(Color.Transparent),
+                    viewModel = listViewModel,
+                    onThreadClick = { message ->
+                        composerViewModel.setMessageMode(Thread(message))
```

Comment:
Again, this is C/P and we don't really need or have threads in Livestream apps. We should remove this.